web/src/message_view.ts

```diff
@@ -1397,7 +1397,7 @@ function handle_post_view_change(
     } else if (filter.is_conversation_view() || filter.includes_full_stream_history()) {
         compose_closed_ui.update_buttons_for_stream_views();
     } else {
-        compose_closed_ui.update_buttons_for_non_specific_views();
+        compose_closed_ui.update_buttons_for_different_views();
```

Comment:
It's not correct to replace `update_buttons_for_non_specific_views` here. Here we need to update the compose state based on the selected message. So, we should condense if conditions to just check if message feed is visible or not and then update the compose box based on which message is selected and update `data-conversation-type` too.